### PR TITLE
Adapted doc to fit new interfaces and extended for media data-provider

### PR DESCRIPTION
**Pull request by wachterjohannes**

### CODE REVIEW

File: cookbook/smart-content-data-provider.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ GET parameters from the website. This can be handled separately from the
 admin-selected. Also different operators (websiteTagsOperator and
 websiteCategoryOperator) are available.
 
-Additional features, which can be provider with a DataProvider:
+Additional features, which can be provided with an DataProvider:
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: It's still `a DataProvider` :wink:

File: cookbook/smart-content-data-provider.rst [hunk 2]
```diff
@@ -98,6 +98,28 @@ The rest of the functionality and Query generation is done in the Trait.
...
+There are following hooks to influence the query generation. These are functions
+which are optional to override in the repository.
+
+.. list-table::
+    :header-rows: 1
+
+    * - Name
+      - Description
+    * - append(QueryBuilder $queryBuilder, $alias, $locale, $options = [])
+      - Additional select, where or joins can be added to the query to match
```
Comment (danrot) [inline cookbook/smart-content-data-provider.rst]: put `select`, `where` and `joins` in backticks to separate them from the rest of the text.

File: reference/content-types/smart_content.rst [hunk 3]
```diff
@@ -303,65 +183,37 @@ Alias: "account"
...
-      - Property of the account.
-    * - urls
-      - array
-      - Property of the account.
-    * - tags
-      - array
-      - Property of the account.
-    * - categories
-      - array
-      - Property of the account.
```
Comment (danrot) [inline reference/content-types/smart_content.rst]: Where is all this information removed here now?

Comment (wachterjohannes) [inline reference/content-types/smart_content.rst]: this is very hard to update all the time i have added the hint that the developer should take a look in the dump of the data.

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**